### ISSUE TITLE
Can not slice

### ISSUE THREAD

**Issue by HJX-zhanS** [joernio/joern#4459]
When I try to use `joern-slice data-flow --sink-filter exec\([\S\s]+\) cpg.bin` for slicing, I encounter an error `WARN DataFlowSlicing$: Exception encountered during slicing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null`

Here is the source code:

```
var exec = require('child_process').exec
  , path = require('path')
  ;


/**
 * open a file or uri using the default application for the file type.
 *
 * @return {ChildProcess} - the child process object.
 * @param {string} target - the file/uri to open.
 * @param {string} appName - (optional) the application to be used to open the
 *      file (for example, "chrome", "firefox")
 * @param {function(Error)} callback - called with null on success, or
 *      an error object that contains a property 'code' with the exit
 *      code of the process.
 */

module.exports = open;

function open(target, appName, callback) {
  var opener;

  if (typeof(appName) === 'function') {
    callback = appName;
    appName = null;
  }

  switch (process.platform) {
  case 'darwin':
    if (appName) {
      opener = 'open -a "' + escape(appName) + '"';
    } else {
      opener = 'open';
    }
    break;
  case 'win32':
    // if the first parameter to start is quoted, it uses that as the title
    // so we pass a blank title so we can quote the file we are opening
    if (appName) {
      opener = 'start "" "' + escape(appName) + '"';
    } else {
      opener = 'start ""';
    }
    break;
  default:
    if (appName) {
      opener = escape(appName);
    } else {
      // use Portlands xdg-open everywhere else
      opener = path.join(__dirname, './xdg-open');
    }
    break;
  }

  if (process.env.SUDO_USER) {
    opener = 'sudo -u ' + process.env.SUDO_USER + ' ' + opener;
  }
  return exec(opener + ' "' + escape(target) + '"', callback);
}

function escape(s) {
  return s.replace(/"/g, '\\\"');
}
```

I want get the slice of `exec(opener + ' "' + escape(target) + '"', callback)`.

Someone can help me?

Thank you very much!

**Comment by DavidBakerEffendi** [comment]
Upon investigating, it seems that there are `.code` properties with `null` values inside. Confirming now the locations of these and if it's the frontend or something with ODB

**Comment by DavidBakerEffendi** [comment]
At the point of the exception, I've obtained the culprit with 
```scala
          c.argument.filter(_.code == null).foreach { n =>
            println(s"${n.id}:${n.method.fullName}:${n.lineNumber.getOrElse(-1)}")
          }
```
Looks like this is the `open` method declaration: 
```
Node ID: 151
Parent Method Full Name: foo.js::program
Line Number: 20
```
However, when jumping into Joern, the code property is non-null
```scala
joern> cpg.graph.node(151)
val res1: overflowdb.Node = MethodRef(
  id = 151L,
  argumentIndex = 2,
  argumentName = None,
  code = "open",
  columnNumber = Some(value = 0),
  dynamicTypeHintFullName = ArraySeq(),
  lineNumber = Some(value = 20),
  methodFullName = "foo.js::program:open",
  order = 2,
  possibleTypes = ArraySeq(),
  typeFullName = "foo.js::program:open"
)
```
Bizzare. Let me try with more memory, as this may be a ODB issue.

**Comment by DavidBakerEffendi** [comment]
Another culprit
```scala
joern> cpg.graph.node(154)
val res1: overflowdb.Node = MethodRef(
  id = 154L,
  argumentIndex = 2,
  argumentName = None,
  code = "escape",
  columnNumber = Some(value = 0),
  dynamicTypeHintFullName = ArraySeq(),
  lineNumber = Some(value = 61),
  methodFullName = "foo.js::program:escape",
  order = 2,
  possibleTypes = ArraySeq(),
  typeFullName = "foo.js::program:escape"
)
```

**Comment by DavidBakerEffendi** [comment]
Ok, seems like it comes down to a concurrency issue when reading code from `MethodRef` nodes when they're instances of `Expression` nodes. @HJX-zhanS a workaround for now would be adding `-p 1` to set parallelism to 1.

**Comment by HJX-zhanS** [comment]
> Ok, seems like it comes down to a concurrency issue when reading code from `MethodRef` nodes when they're instances of `Expression` nodes. @HJX-zhanS a workaround for now would be adding `-p 1` to set parallelism to 1.

Thank you, DavidBakerEffendi. When I use `-p 1`, the above error no longer appear. However, I think there may be some issues. Because when I set `--sink-filter "exec\([\S\s]+?\)"`  to try to get the slice of `exec(opener + ' "' + escape(target) + '"', callback)`, the result is empty. When I set `--sink-filter "exec"`, I can get a slice. However, the slice is only related to `var exec = require('child_process').exec`.
